doc/source/programs/gdal_raster_footprint.rst

```diff
@@ -0,0 +1,175 @@
+.. _gdal_raster_footprint_subcommand:
+
+================================================================================
+"gdal raster footprint" sub-command
+================================================================================
+
+.. versionadded:: 3.11
+
+.. only:: html
+
+    Compute the footprint of a raster dataset
+
+.. Index:: gdal raster footprint
+
+Synopsis
+--------
+
+.. program-output:: gdal raster footprint --help-doc
+
+Description
+-----------
+
+:program:`gdal raster footprint` computes the footprint of
+a raster file, taking into account nodata values (or more generally the mask
+band attached to the raster bands), and generating polygons/multipolygons
+corresponding to areas where pixels are valid, and write to an output vector file.
+
+.. TODO: uncomment when available
+.. The :program:`gdal raster clean-collar` utility may be run as a pre-processing
+.. step to generate proper mask bands.
+
+The following options are available:
+
+.. include:: gdal_options/of_vector.rst
+
+.. include:: gdal_options/co.rst
+
+.. include:: gdal_options/lco.rst
+
+.. option:: --output-layer <OUTPUT-LAYER>
+
+    Output layer name. Defaults to ``footprint``.
+
+.. include:: gdal_options/overwrite.rst
+
+.. option:: --append
+
+    Whether appending features to existing layer(s) is allowed
+
+.. option:: -b <band>
+
+    Select an input <band> to be processed. Bands are numbered from 1.
+    Multiple :option:`-b` switches may be used to select a set of input bands.
+    If not specified, all bands are taken into account. The way multiple bands
+    are combined is controlled by :option:`--combine-bands`
+
+.. option:: --combine-bands union|intersection
+
+    Defines how the mask bands of the selected bands are combined to generate
+    a single mask band, before being vectorized.
+    The default value is ``union``: that is a pixel is valid if it is valid at least
+    for one of the selected bands.
+    ``intersection`` means that a pixel is valid only if it is valid for all
+    selected bands.
+
+.. option:: --ovr <index>
+
+   To specify which overview level of source file must be used, when overviews
+   are available on the source raster. By default the full resolution level is
+   used. The index is 0-based, that is 0 means the first overview level.
+   This option is mutually exclusive with :option:`--srcnodata`.
+
+.. option:: --srcnodata <value>
+
+    Set nodata values for input bands (different values can be supplied for each band).
```

Comment:
There is no test for different nodata values for each band.

Reply:
Good catch! test added